### PR TITLE
moserial: init at 3.0.21

### PR DESCRIPTION
**Pull request by linsui**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/moserial/default.nix [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "moserial";
+  version = "3.0.21";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "GNOME";
+    repo = pname;
+    rev = "moserial_${builtins.replaceStrings [ "." ] [ "_" ] version}";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/moserial/default.nix:24]: This should be lib.replaceStringd

File: pkgs/tools/misc/moserial/default.nix [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  };
+
+  buildInputs = [
+    glib
+    graphviz
+    yelp-tools
+    gtk3
+  ];
+
+  nativeBuildInputs = [
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/moserial/default.nix:35]: Native is ordered before buildInputs.

File: pkgs/tools/misc/moserial/default.nix [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    )
+  '';
+
+  meta = with lib; {
+    description = "clean, friendly gtk-based serial terminal for the gnome desktop";
+    homepage = "https://wiki.gnome.org/moserial";
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [ linsui ];
+    platforms = platforms.linux;
+    mainProgram = "moserial";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/moserial/default.nix:55]: That is the default.

File: pkgs/tools/misc/moserial/default.nix [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  ];
+
+  preFixup = ''
+    gappsWrapperArgs+=(
+      --prefix PATH : ${lib.makeBinPath [ lrzsz ]}
+    )
+  '';
+
+  meta = with lib; {
+    description = "clean, friendly gtk-based serial terminal for the gnome desktop";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/moserial/default.nix:50]: *Clean

**Review by dbirks — APPROVED** [review]
Thanks for this! Just built and used it :+1: 

Result of `nixpkgs-review pr 164267` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>moserial</li>
  </ul>
</details>

**Pull request merged**